### ISSUE TITLE
Fail to Compile Against OpenCL 1.1 Headers

### ISSUE THREAD

**Issue by thibaudh** [mikeseven/node-webcl#2]
It looks like some #ifdefs are missing:

```
In file included from ../src/cl_checks.cc:1:0:
../src/cl_checks.h: In function ‘bool webcl::validateMemFlags(int)’:
../src/cl_checks.h:19:46: error: ‘CL_MEM_HOST_NO_ACCESS’ was not declared in this scope
   return (value>=CL_MEM_READ_WRITE && value<=CL_MEM_HOST_NO_ACCESS && value!=(1<<6));
                                              ^
../src/cl_checks.cc: In function ‘int webcl::getChannelCount(int)’:
../src/cl_checks.cc:147:10: error: ‘CL_DEPTH’ was not declared in this scope
     case CL_DEPTH:
          ^
../src/cl_checks.cc:148:10: error: ‘CL_DEPTH_STENCIL’ was not declared in this scope
     case CL_DEPTH_STENCIL:
          ^
make: *** [Release/obj.target/webcl/src/cl_checks.o] Error 1
```

**Comment by mikeseven** [comment]
This looks like you have outdated OpenCL headers in your system.
Make sure you have OpenCL 1.2 headers.

-- Mike

From:  Thibaud notifications@github.com
Reply-To:  mikeseven/node-webcl
<reply+000c582ca68ea55fb5041a47d7934fccc7856a6d654e008392cf0000000110e373b39
2a169ce03575240@reply.github.com>
Date:  Friday, January 30, 2015 at 8:30 AM
To:  mikeseven/node-webcl node-webcl@noreply.github.com
Subject:  [node-webcl] Fail to Compile Against OpenCL 1.1 Headers (#2)

It looks like some #ifdefs are missing:
In file included from ../src/cl_checks.cc:1:0:
../src/cl_checks.h: In function Œbool webcl::validateMemFlags(int)¹:
../src/cl_checks.h:19:46: error: ŒCL_MEM_HOST_NO_ACCESS¹ was not declared in
this scope
   return (value>=CL_MEM_READ_WRITE && value<=CL_MEM_HOST_NO_ACCESS &&
value!=(1<<6));
                                              ^
../src/cl_checks.cc: In function Œint webcl::getChannelCount(int)¹:
../src/cl_checks.cc:147:10: error: ŒCL_DEPTH¹ was not declared in this scope
     case CL_DEPTH:
          ^
../src/cl_checks.cc:148:10: error: ŒCL_DEPTH_STENCIL¹ was not declared in
this scope
     case CL_DEPTH_STENCIL:
          ^
make: *** [Release/obj.target/webcl/src/cl_checks.o] Error 1
‹
Reply to this email directly or view it on GitHub
<https://github.com/mikeseven/node-webcl/issues/2> .

**Comment by thibaudh** [comment]
Yes I have OpenCL 1.1 headers. This is all we get on Nvidia platforms. The README claims it is supported: `This is an implementation of Khronos WebCL specification using NodeJS. This implementation solely relies on OpenCL 1.1 C headers.`

**Comment by mikeseven** [comment]
Ah I see. I upgraded to OpenCL 1.2 since all my machines here are now even
2.0 ready.
OK I need to add some #ifdef

-- Mike

From:  Thibaud notifications@github.com
Reply-To:  mikeseven/node-webcl
<reply+000c582cef3a27f6a4b2fd917c09ba6bdda0f31d0051d1df92cf0000000110e3796e9
2a169ce03575240@reply.github.com>
Date:  Friday, January 30, 2015 at 8:55 AM
To:  mikeseven/node-webcl node-webcl@noreply.github.com
Cc:  Mikaël Bourges-Sévenier mikeseven@gmail.com
Subject:  Re: [node-webcl] Fail to Compile Against OpenCL 1.1 Headers (#2)

Yes I have OpenCL 1.1 headers. This is all we get on Nvidia platforms. The
README claims it is supported: This is an implementation of Khronos WebCL
specification using NodeJS. This implementation solely relies on OpenCL 1.1
C headers.

‹
Reply to this email directly or view it on GitHub
https://github.com/mikeseven/node-webcl/issues/2#issuecomment-72232368 .

**Comment by thibaudh** [comment]
I will be happy to help if you want to test a patch :)

**Comment by anuragingole90** [comment]
I am also having same problem. 
Is it resolved?

**Comment by mikeseven** [comment]
sure, thanks!

— mike

**Comment by mfwei** [comment]
I still have the same problem - fail to compile with OpenCL 1.1 headers
How to solve it ? Thanks a lot !
